src/olympia/blocklist/tests/test_mlbf.py

```diff
@@ -193,6 +214,47 @@ def test_get_data_from_db(self):
         mlbf_data = MLBFDataBaseLoader(mlbf.storage)
         assert mlbf.data._raw == mlbf_data._raw
 
+    def test_cache_json_is_sorted(self):
```

Comment:
I don't think this test is quite right because if I apply it to current `master`, it passes.

Reply:
If you change the order by in the queries the test fails.

Reply:
Do the queries on `master` have ORDER BY clauses? The test doesn't capture the "broken" behavior that this PR is supposed to fix. I would expect a failing test on `master` and a passing test with this PR.

Reply:
They do not.. and for me at least it fails on master. It could be that this test is actually flaky on master.. if we don't specify ordering it _could_ be returned in the right order and the fewer number of items in the list the higher the probability would be.

This is from latest master right now.

<img width="1287" alt="image" src="https://github.com/user-attachments/assets/3cf9c678-f196-4b6c-8d80-bb47b64e347d">

Reply:
Currently running a 100 loop on it and it is indeed flaky. Sometimes passes.